app/controllers/admin/product_import_controller.rb

```diff
@@ -0,0 +1,52 @@
+require 'roo'
+
+class Admin::ProductImportController < Spree::Admin::BaseController
+
+  before_filter :check_upload, except: :index
+
+  def import
+    # Save uploaded file to tmp directory
+    @filepath = save_upload(params[:file])
+    @importer = ProductImporter.new(File.new(@filepath), editable_enterprises)
+
+    if @importer.errors.present?
+      flash[:notice] = @importer.errors.full_messages.to_sentence
+    end
+  end
+
+  def save
+    file = File.new(params[:filepath])
```

Comment:
@oeoeaio Agreed, passing in the filepath is not good. Not sure a new model is needed, but I'll take another look at it. Also, the imported file gets deleted once the data is saved successfully, but if the user imports a spreadsheet then leaves without completing it the file stays on the server. Currently it doesn't get cleaned up or deleted, which could lead to junk spreadsheets sitting around in the filesystem...